packages/salesforce-adapter/src/transformers/transformer.ts

```diff
@@ -1137,13 +1142,27 @@ export const getSObjectFieldElement = (
   return new Field(parent, fieldName, naclFieldType, annotations)
 }
 
+export const toDeployableInstance = (element: InstanceElement): InstanceElement => {
+  const removeLocalOnly: TransformFunc = ({ value, field }) => (
+    (field !== undefined && field.annotations[FIELD_ANNOTATIONS.LOCAL_ONLY] === true)
```

Comment:
local nit:
```suggestion
    field?.annotations?.[FIELD_ANNOTATIONS.LOCAL_ONLY] === true
```
but maybe worth extracting this to a function `isLocalOnly` or something, not sure